File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> result;
        int product;
        for(int i = 0; i < nums.size(); i++) {
            product = nums[0];
            for(int j = 1; j < nums.size(); j++) {
                if(nums[i] == 0 && i == j) {
                    product = product;
                } else {
                    product *= nums[j];
                }
            }

            if(nums[i] == 0) {
                result.push_back(product);
            } else {
                product /= nums[i];
                result.push_back(product);
            }
        }
        return result;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts = {}
        frequencies = [[] for i in range(len(nums) + 1)]

        for num in nums:
            counts[num] = counts.get(num, 0) + 1

        for num in counts.keys():
            frequencies[counts[num]].append(num)

        result = []
        for freq in frequencies[::-1]:
            if len(freq) != 0:
                for n in freq:
                    result.append(n)
                    if len(result) == k:
                        return result
        return []